docs/zkapps/o1js/common-types-and-functions.md

```diff
@@ -0,0 +1,354 @@
+
+<!-- prettier-ignore -->
+```ts
+const x = Provable.if(
+  Bool(foo), // The condition to evaluate inside if, must be of type Bool
+  a, // You can return any provable type from Provable.if
+  b // a and b must be of the same provable type
+); // Behaves like `foo ? a : b`
+```
+
+`Provable.if()` allows you to include any logical computation you need inside your ZKP. There is nothing wrong with evaluating more complex expressions inside `Provable.if()` and returning the final result.
+
+However, it works differently from ternary statements in the other languages and frameworks. Both arguments of `Provable.if()` _always_ executes, regardless of the truth value of the condition. `Provable.if()` choses which one to return based on the truth value and sets the new variable. This is why you should never assign anything outside of the if scope or assert anything.
+
+<!-- prettier-ignore -->
+```ts
+const x = Provable.if(
+  Bool(foo),
+  (() => { // Here, we use arrow functions to return a Field element, but you can use any function that returns a provable type
+    return Field(5).div(3);
+  })(),
+  (() => {
+    return Field(5).div(2);
+  })()
+); // This variable is of type Field
+```
+
+# Loops
+
+Loops also work differently in o1js because of the same reason with conditionals. In a mathematical statement, you cannot chose how many times to do something based on a variable that is not known at the compile time. Similarly, o1js does not allow you to use dynamically sized loops, loops that the iteration count is not known at the compilation.
+
+<!-- prettier-ignore -->
+```ts
+// This will NOT work
+while (foo) {
+  x.add(y);
+}
+```
+
+You can overcome this limitation by using static sized loops that iterate always the same amount of time. You can perform provable operations with these loops using the `Provable.if`.
+
+<!-- prettier-ignore -->
+```ts
+function powerOfTwo(pow: Field): Field {
+  const MAX_ITERATION_COUNT = 64;
+
+  pow.assertGreaterThanOrEqual(0);
+  pow.assertSmallerThanOrEqual(MAX_ITERATION_COUNT); // This code is to calculate up to 2^64, you can optimize this based on your needs
+
+  let result = Field(1); // Result of the calculation
+  let power = Field(0); // Current power of the iteration
+  let loopEnded = Bool(false); // Checking if the loop has ended yet
+
+  for (let i = 0; i <= MAX_ITERATION_COUNT; i++) { // Static sized loop
+    loopEnded = Provable.if( // End Condition
+      Field(i).greaterThanOrEqual(power),
+      Bool(true),
+      Bool(false)
+    );
+    result = result.mul(Provable.if( // Result
+      loopEnded,
+      Field(1),
+      Field(2)
+    ));
+    power = power.add(Provable.if( // Power
+      loopEnded,
+      Field(0),
+      Field(1)
+    ));
+  };
+
+  return result;
+};
+```
+
+# Functions
+
+Functions work as you would expect in TS. However, your functions inside ZKPs must take provable arguments and return provable types, just like anything in o1js. For example:
```

Comment:
That's not correct. It's common for functions to take parameters that define the circuit, not only variables.